data/src/main/java/com/example/data/backendapi/dao/OAuth.java

```diff
@@ -0,0 +1,40 @@
+package com.example.data.backendapi.dao;
+
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import com.example.data.backendapi.dto.Customer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Repository;
+
+//TODO:user's roles ENUM object will be added
+@Repository
+public class OAuth {
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    public Customer getUserDetails(String username) {
+        Collection<GrantedAuthority> grantedAuthoritiesList = new ArrayList<>();
+        String userSQLQuery = "SELECT * FROM USERS WHERE USERNAME=?";
+        List<Customer> list = jdbcTemplate.query(userSQLQuery, new String[] { username },
+                (ResultSet rs, int rowNum) -> {
+
+                    Customer user = new Customer();
+                    user.setUsername(username);
+                    user.setPassword(rs.getString("PASSWORD"));
+                    return user;
+                });
+        if (list.size() > 0) {
+            GrantedAuthority grantedAuthority = new SimpleGrantedAuthority("ROLE_SYSTEMADMIN");
+            grantedAuthoritiesList.add(grantedAuthority);
+            list.get(0).setGrantedAuthoritiesList(grantedAuthoritiesList);
+            return list.get(0);
+        }
+        return null;
```

Comment:
Try to avoid `null` consider to use `java.util.Optional` instead if you need to return no object.